pkg/downloader/chart_downloader.go

```diff
@@ -80,13 +80,24 @@ type ChartDownloader struct {
 //
 // Returns a string path to the location where the file was downloaded and a verification
 // (if provenance was verified), or an error if something bad happened.
-func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *provenance.Verification, error) {
-	u, g, err := c.ResolveChartVersion(ref, version)
+func (c *ChartDownloader) DownloadTo(ref, username, password, version, dest string) (string, *provenance.Verification, error) {
```

Comment:
I feel like these should be fields of the ChartDownloader rather than fields in DownloadTo. That way the functions are backwards compatible for third party clients using pkg/downloader in their projects.

Reply:
Done.